url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/GadajAsterisk.java

```diff
@@ -0,0 +1,695 @@
+						sleep(100);
+					
+				} catch (InterruptedException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
+				}
+			}
+		}
+		} catch (Exception e) {
+			e.printStackTrace();
+			System.exit(-1);
+		}
+	}
+
+	public void sendMessage(AgentConnection agent, String outputLine) {
+		sendMessage(agent, outputLine, true);
+	}
+
+	public void sendMessage(AgentConnection agent, String outputLine,
+			boolean printConfirmation) {
+		WiadomoscOdAgenta w;
+		Date time = GregorianCalendar.getInstance().getTime();
+
+		// obsługa kolumn z wieloma argumentami typu: EVENT [ProxyClient], Action [Pause], Value [True], Type [Default], Agent [313], Exten [874]
+		Command c = Command.factory(outputLine, null);
+		
+		if  (! (c instanceof CommandUnknown)){
+			p.log.info("We got NG command: "+c.getClass()+ " | "+c.getOrginalLine());
+			if (c instanceof EventProxyClientPause || c instanceof EventProxyClientCallTag){
+				if (c instanceof EventProxyClientPause)  //supervisor pauzować może nie siebie
+					agent=p.agenci.getAgentConnectionByNumer(((EventProxyClientPause)c).getAgent().getId().toString()); 
+				w = new WiadomoscOdAgenta(agent, numer++, time, c);
+				try{
+					agentListLock.lock();
+					wiadomosciOdAgenta.add(w);
+				} finally{
+					agentListLock.unlock();
+				}
+			} else {
+				p.log.warn("I can't process: "+c.getClass());
+			}
+			return;
+		}
+		
+		// tu po staremu
+		outputLine = outputLine.replaceAll("\\s", "");
+		
+		// komendy z argumentami
+		if (outputLine.contains(":")) {
+			p.log.info("+INFO : command '" + outputLine);
+			String sline[] = outputLine.split(":");
+			if (sline.length == 2) {
+				w = new WiadomoscOdAgenta(agent, sline[0], sline[1], numer++);
+				w.time = time;
+				try{
+					agentListLock.lock();
+					wiadomosciOdAgenta.add(w);
+				} finally{
+					agentListLock.unlock();
+				}
+				if (printConfirmation)
+					agent.out.println("+OK '" + outputLine + "'"); // FIXME
+																	// dorobic
+																	// sprawdzanie,
+																	// czy jest
+																	// taka
+																	// wiadomosc
+				return;
+			} else {
+  
+				
+			}
+		}else{
+		// komendy bezargumentowe
+		if (Commands.COMMANDS.contains(outputLine)) {
+			p.log.info("+INFO New command '" + outputLine + "' from agent: "
+					+ agent.agentNumber + ", from IP: " + agent.ip + ":"
+					+ agent.port + ", num: "+(numer+1));
+			if (printConfirmation)
+				agent.out.println("+OK '" + outputLine + "'");
+			w = new WiadomoscOdAgenta(agent, outputLine, numer++);
+			w.time = time;
+			try{
+				agentListLock.lock();
+				wiadomosciOdAgenta.add(w);
+			} finally{
+				agentListLock.unlock();
+			}
+		} else {
+			p.log.info("+INFO Unknown command '" + outputLine
+					+ "' from agent: " + agent.agentNumber + ", from IP: "
+					+ agent.ip);
+			if (printConfirmation)
+				agent.out.println("+ERR No command '" + outputLine + "'");
+		}
+		}
+	}
+
+	public boolean wyslijWiadomosci() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 151 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.